### PR TITLE
Refactoring + Test Fix

### LINKED ISSUE #132: Fix async test cases for express tooling
Fix problem described in https://github.com/zalando-incubator/authmosphere/pull/130/files/a35013288e9e091795d870f92a7b84ffa329d061#diff-c0d33e3fef66f9e1990b28d8d4eaca04R73

### PR DESCRIPTION
**Pull request by ISO50**
close #132

### CODE REVIEW

File: src/express-tooling.ts [hunk 1]
```diff
@@ -64,15 +72,14 @@ function requireScopesMiddleware(scopes: string[],
         } catch (e) {
           safeLogger(logger).error(`Error while executing precedenceErrorHandler: ${e}`);
         }
-      });
+      })
+      // even if precedenceFunction and precedenceErrorHandler throws
+      // fallback to the default way
+      .catch(() => acceptOrRejectRequest(req, res, next, scopes, logger));
```
Comment (Retro64) [inline src/express-tooling.ts]: unreachable

File: src/express-tooling.ts [hunk 2]
```diff
@@ -89,8 +96,9 @@ function requireScopesMiddleware(scopes: string[],
  * @param logger - optional logger
  * @returns express middleware
  */
-function handleOAuthRequestMiddleware(options: MiddlewareOptions,
-                                      logger?: Logger): RequestHandler {
+type handleOAuthRequestMiddleware = (options: MiddlewareOptions, logger?: Logger) => RequestHandler
+const handleOAuthRequestMiddleware: handleOAuthRequestMiddleware =
+  (options, logger): RequestHandler => {
```
Comment (Retro64) [inline src/express-tooling.ts]: RequestHandler might be dropped

File: README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # authmosphere {🌍}
 
 [![Build Status](https://travis-ci.org/zalando-incubator/authmosphere.svg)](https://travis-ci.org/zalando-incubator/authmosphere)
-[![Coverage Status](https://coveralls.io/repos/github/zalando-incubator/authmosphere/badge.svg)](https://coveralls.io/github/zalando-incubator/authmosphere)
+[![Coverage Status](https://coveralls.io/repos/github/zalando-incubator/authmosphere/badge.svg?branch=master)](https://coveralls.io/github/zalando-incubator/authmosphere)
```
Comment (Retro64) [inline README.md]: Wrong line - build status should have been changed

**Comment by Retro64** [comment]
:+1:

**Comment by bzums** [comment]
👍

**Comment by Retro64** [comment]
:+1:

**Pull request merged**